aptos-move/move-examples/staking/sources/commission.move

```diff
@@ -0,0 +1,145 @@
+        /// This can happen if the commission rate is set too high or APT price is too low.
+        commission_debt: u64,
+    }
+
+    fun init_module(commission_signer: &signer) {
+        let signer_cap = resource_account::retrieve_resource_account_cap(commission_signer, @manager);
+        move_to(commission_signer, CommissionConfig {
+            manager: @manager,
+            operator: @operator,
+            yearly_commission_amount: INITIAL_COMMISSION_AMOUNT,
+            signer_cap,
+            last_update_secs: timestamp::now_seconds(),
+            commission_debt: 0,
+        });
+    }
+
+    #[view]
+    public fun operator(): address acquires CommissionConfig {
+        CommissionConfig[@staking].operator
+    }
+
+    #[view]
+    public fun yearly_commission_amount(): u64 acquires CommissionConfig {
+        CommissionConfig[@staking].yearly_commission_amount
+    }
+
+    #[view]
+    public fun commission_owed(): u64 acquires CommissionConfig {
+        let config = &CommissionConfig[@staking];
+        // Commission earned so far = per second commission rate * seconds passed.
+        let now_secs = timestamp::now_seconds();
+        let seconds_passed = now_secs - config.last_update_secs;
+        let commission_earned = math64::mul_div(seconds_passed, config.yearly_commission_amount, ONE_YEAR_IN_SECONDS);
+
+        commission_earned + config.commission_debt
+    }
+
+    /// Can be called by the manager to change the yearly commission amount.
+    public entry fun set_yearly_commission_amount(manager: &signer, new_commission: u64) acquires CommissionConfig {
+        assert_manager(manager);
+        CommissionConfig[@staking].yearly_commission_amount = new_commission;
+    }
+
+    /// Can be called by the manager or operator to change the account that receives the commission.
+    public entry fun set_operator(account: &signer, new_operator: address) acquires CommissionConfig {
+        assert_manager_or_operator(account);
+        CommissionConfig[@staking].operator = new_operator;
+    }
+
+    /// Distribute the commission to operator and remaining amount to manager.
+    /// Can only be called by the manager or operator.
+    public entry fun distribute_commission(account: &signer) acquires CommissionConfig {
+        assert_manager_or_operator(account);
+
+        // Commission owed so far plus any debt.
+        let commission_owed = commission_owed() as u128;
+
+        // Only manager or operator can call this function.
+        let config = &mut CommissionConfig[@staking];
+        config.last_update_secs = timestamp::now_seconds();
+        // Commission debt is already included in commission_owed by the commission_owed function.
+        config.commission_debt = 0;
+
+        // Commission in APT = commission earned / APT price.
+        let apt_price = oracle::get_apt_price();
+        let commission_in_apt = math128::mul_div(commission_owed, oracle::precision(), apt_price) as u64;
```

Comment:
do we need another `precision()` here as `commission_owed` is in octas?

Reply:
Good catch. The price pyth returns is for APT not octas. I've updated the code to take this into consideration. Also updated the unit tests with real number returned from the Pyth oracle to ensure they're realistic and correct